plugin/pkg/admission/concurrentupdate/admission_test.go

```diff
@@ -0,0 +1,259 @@
+// the specified ServiceInstance resources
+func newFakeServiceCatalogClientWithInstances(instances ...servicecatalog.ServiceInstance) *fake.Clientset {
+	fakeClient := &fake.Clientset{}
+
+	instanceList := &servicecatalog.ServiceInstanceList{
+		ListMeta: metav1.ListMeta{
+			ResourceVersion: "1",
+		},
+	}
+	instanceList.Items = append(instanceList.Items, instances...)
+
+	fakeClient.AddReactor("list", "serviceinstances", func(action core.Action) (bool, runtime.Object, error) {
+		return true, instanceList, nil
+	})
+
+	return fakeClient
+}
+
+// newFakeServiceCatalogClientWithInstanceCredentials creates a fake clientset
+// that provides the specified ServiceInstanceCrendential resources
+func newFakeServiceCatalogClientWithInstanceCredentials(instanceCredentials ...servicecatalog.ServiceInstanceCredential) *fake.Clientset {
+	fakeClient := &fake.Clientset{}
+
+	instanceCredentialList := &servicecatalog.ServiceInstanceCredentialList{
+		ListMeta: metav1.ListMeta{
+			ResourceVersion: "1",
+		},
+	}
+	instanceCredentialList.Items = append(instanceCredentialList.Items, instanceCredentials...)
+
+	fakeClient.AddReactor("list", "serviceinstancecredentials", func(action core.Action) (bool, runtime.Object, error) {
+		return true, instanceCredentialList, nil
+	})
+
+	return fakeClient
+}
+
+func getTestServiceInstance() servicecatalog.ServiceInstance {
+	return servicecatalog.ServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       "test-instance",
+			Namespace:  "test-namespace",
+			Generation: 1,
+		},
+		Status: servicecatalog.ServiceInstanceStatus{
+			ReconciledGeneration: 1,
+		},
+	}
+}
+
+func getTestServiceInstanceCredential() servicecatalog.ServiceInstanceCredential {
+	return servicecatalog.ServiceInstanceCredential{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       "test-instance",
+			Namespace:  "test-namespace",
+			Generation: 1,
+		},
+		Status: servicecatalog.ServiceInstanceCredentialStatus{
+			ReconciledGeneration: 1,
+		},
+	}
+}
+
+func TestAdmitInstance(t *testing.T) {
+	cases := []struct {
+		name              string
+		oldInstance       servicecatalog.ServiceInstance
+		newInstance       servicecatalog.ServiceInstance
+		subresource       string
+		expectedErrorCode int32
+	}{
+		{
+			name:        "no pending change",
+			oldInstance: getTestServiceInstance(),
+			newInstance: getTestServiceInstance(),
+		},
+		{
+			name:        "pending update",
+			oldInstance: getTestServiceInstance(),
+			newInstance: func() servicecatalog.ServiceInstance {
+				i := getTestServiceInstance()
+				i.Generation = 2
+				return i
+			}(),
+			expectedErrorCode: 409,
+		},
+		{
+			name:        "pending delete",
+			oldInstance: getTestServiceInstance(),
+			newInstance: func() servicecatalog.ServiceInstance {
+				i := getTestServiceInstance()
+				now := metav1.Now()
+				i.DeletionTimestamp = &now
+				return i
+			}(),
+			expectedErrorCode: 409,
+		},
+		{
+			name:        "status update",
```

Comment:
we should have a case where status is updated when current and reconciled generation match

Reply:
Interesting. I definitely wrote that test. Don't know where it went. Wonder what else I managed to lose.